src/bicep/examples/automationAccount/automationAccount.bicep

```diff
@@ -0,0 +1,64 @@
+/*
+Deployes an automation account with modules/runbook/diagnostic logs
+Makes use of example azure automation bicep: https://github.com/Azure/bicep/tree/main/docs/examples/301/automation-account-import-runbooks-and-modules
+*/
+targetScope = 'subscription'
+
+param mlzDeploymentVariables object = json(loadTextContent('../deploymentVariables.json'))
+
+@description('The name of the automation account which will be created.  If unchanged or not specified, the MLZ resource prefix + "-aAc" will be utilized.')
+param automationAcctName string = '${mlzDeploymentVariables.mlzResourcePrefix.Value}-aAc'
+
+@description('The name of the resource group in which the automation account will be deployed. If unchanged or not specified, the MLZ operations resource group is used.')
+param targetResourceGroup string = '${mlzDeploymentVariables.spokes.Value[1].resourceGroupName}'
+
+@description('The URL location to the powershell runbook you wish to use')
+param automationRunbook string = 'https://raw.githubusercontent.com/Azure/missionlz/breestryker/AzAutomationBicepEx/src/bicep/examples/automationAccount/test.ps1'
+
+@description('A string dictionary of tags to add to deployed resources. See https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/tag-resources?tabs=json#arm-templates for valid settings.')
+param tags object = {}
+var defaultTags = {
+  'DeploymentType': 'MissionLandingZoneARM'
+}
+var calculatedTags = union(tags, defaultTags)
+
+var targetSubscriptionId_Var = targetResourceGroup == '${mlzDeploymentVariables.spokes.Value[1].resourceGroupName}' ? '${mlzDeploymentVariables.spokes.Value[1].subscriptionId}' : subscription().subscriptionId
+var location = deployment().location
```

Comment:
The current version of Bicep emits this warning for this line:

```plaintext
automationAccount.bicep(26,16) : Warning no-loc-expr-outside-params: Use a parameter here instead of 'deployment().location'. 'resourceGroup().location' and 'deployment().location' should only be used as a default value for parameters. [https://aka.ms/bicep/linter/no-loc-expr-outside-params]
```

Reply:
What we ended up doing in the other modules reporting this was a warning was defaulting a param like:

https://github.com/Azure/missionlz/blob/199b986c770d0d331b6d99fce39d9b9807af2e28/src/bicep/modules/policyAssignment.bicep#L15-L16